### PR TITLE
rfq: mimimal viable withdrawal api

### PR DESCRIPTION
**Pull request by trajan0x**
**Description**

Add a withdrawal api to rfq

To do:

- [ ] Add to docs - #2834

Future work:
- [x] Small cli #2826
- [ ] Opbot inclusion w/ rbac? - wontfix

## Summary by CodeRabbit

- **New Features**
  - Added functionality for handling token withdrawals.
  - Introduced command-line interface support for withdrawal operations.
  - Enabled API withdrawals with configuration options for whitelisting addresses.

- **Bug Fixes**
  - Improved handling and validation of withdrawal requests.

- **Tests**
  - Added tests for Ethereum and ERC20 token withdrawals.
  - Introduced tests for token ID decoding functionality.

- **Documentation**
  - Updated comments for configuration fields for clarity.

### CODE REVIEW

File: services/rfq/relayer/relapi/handler.go [hunk 1]
```diff
@@ -128,3 +139,141 @@
...
+	if err := c.ShouldBindJSON(&req); err != nil {
+		c.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
+		return
+	}
+
+	// validate the token address
+	if !tokenIDExists(h.cfg, req.TokenAddress, int(req.ChainID)) {
+		c.JSON(http.StatusBadRequest, gin.H{"error": fmt.Sprintf("invalid token address %s for chain %d", req.TokenAddress.Hex(), req.ChainID)})
+		return
+	}
```
[the remark this answers is not in the document]
Comment (trajan0x) [reply services/rfq/relayer/relapi/handler.go:156]: Sure

File: services/rfq/relayer/relapi/handler.go [hunk 2]
```diff
@@ -128,3 +139,142 @@
...
```
[the remark this answers is not in the document]
Comment (trajan0x) [reply services/rfq/relayer/relapi/handler.go:156]: Sure

**Review by dwasse — APPROVED** [review]

File: services/rfq/relayer/relapi/handler.go [hunk 3]
```diff
@@ -128,3 +139,141 @@
...
+
+	// validate the withdrawal address
+	if !toAddressIsWhitelisted(h.cfg, req.To) {
+		c.JSON(http.StatusBadRequest, gin.H{"error": fmt.Sprintf("withdrawal address %s is not whitelisted", req.To.Hex())})
+		return
+	}
```
[the remark this answers is not in the document]
Comment (trajan0x) [reply services/rfq/relayer/relapi/handler.go:165]: can you generate?

File: services/rfq/relayer/relapi/handler.go [hunk 4]
```diff
@@ -128,3 +139,141 @@
...
+		To           string `json:"to"`
+		*Alias
+	}{
+		Alias: (*Alias)(wr),
+	}
+
+	if err := json.Unmarshal(data, aux); err != nil {
+		//nolint: wrapcheck
+		return err
+	}
```
[the remark this answers is not in the document]
Comment (trajan0x) [reply services/rfq/relayer/relapi/handler.go:282]: generate pls

File: services/rfq/relayer/relconfig/config.go [hunk 5]
```diff
@@ -156,6 +160,26 @@
...
+	if len(split) != 2 {
+		return chainID, addr, fmt.Errorf("invalid token ID: %s", id)
+	}
+	chainID, err = strconv.Atoi(split[0])
+	if err != nil {
+		return chainID, addr, fmt.Errorf("invalid chain ID: %s", split[0])
+	}
+	addr = common.HexToAddress(split[1])
+	return chainID, addr, nil
+}
```
[the remark this answers is not in the document]
Comment (trajan0x) [reply services/rfq/relayer/relconfig/config.go:181]: please generate

File: services/rfq/relayer/relapi/handler.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (trajan0x) [reply services/rfq/relayer/relapi/handler.go:165]: running the whole api for this is crazy. export it

**Pull request merged**